src/main/java/com/google/gcp_variant_transforms/library/VariantToBqUtilsImpl.java

```diff
@@ -0,0 +1,238 @@
+    }
+  }
+
+  public void addGenotypes(TableRow row, List<Allele> alleles,
+                           VariantContext variantContext) {
+    List<Integer> genotypes = new ArrayList<>();
+    for (Allele allele : alleles) {
+      String alleleStr = allele.getDisplayString();
+      if (alleleStr.equals(Constants.MISSING_FIELD_VALUE)) {
+        genotypes.add(Constants.MISSING_GENOTYPE_VALUE);
+      } else {
+        genotypes.add(variantContext.getAlleleIndex(allele));
+      }
+    }
+    row.set(Constants.ColumnKeyConstants.CALLS_GENOTYPE, genotypes);
+  }
+
+  public void addInfoAndPhaseSet(TableRow row, Genotype genotype, VCFHeader vcfHeader) {
+    String phaseSet = "";
+
+    if (genotype.hasAD()) { row.set("AD", genotype.getAD()); }
+    if (genotype.hasDP()) { row.set("DP", genotype.getDP()); }
+    if (genotype.hasGQ()) { row.set("GQ", genotype.getGQ()); }
+    if (genotype.hasPL()) { row.set("PL", genotype.getPL()); }
+    Map<String, Object> extendedAttributes = genotype.getExtendedAttributes();
+    for (String extendedAttr : extendedAttributes.keySet()) {
+      if (extendedAttr.equals(Constants.PHASESET_FORMAT_KEY)) {
+        String phaseSetValue = extendedAttributes.get(Constants.PHASESET_FORMAT_KEY).toString();
+        phaseSet = phaseSetValue.equals(Constants.MISSING_FIELD_VALUE) ? null : phaseSetValue;
+      } else {
+        // The rest of fields need to be converted to the right type by VCFCodec decode function.
+        VCFFormatHeaderLine formatMetadata = vcfHeader.getFormatHeaderLine(extendedAttr);
+        VCFHeaderLineType formatType = formatMetadata.getType();
+        VCFHeaderLineCount formatCountType = formatMetadata.getCountType();
+        if (formatCountType == VCFHeaderLineCount.INTEGER) {
+          row.set(extendedAttr, convertToDefinedType(extendedAttributes.get(extendedAttr),
+              formatType, formatMetadata.getCount()));
+        } else {
+          // If field number in the VCFHeader is ".", should pass a default count
+          row.set(extendedAttr, convertToDefinedType(extendedAttributes.get(extendedAttr),
+              formatType, Constants.DEFAULT_FIELD_COUNT));
+        }
+      }
+    }
+    if (phaseSet == null || phaseSet.length() != 0) {
+      // PhaseSet is presented(MISSING_FIELD_VALUE('.') or real value).
+      row.set(Constants.ColumnKeyConstants.CALLS_PHASESET, phaseSet);
+    } else {
+      row.set(Constants.ColumnKeyConstants.CALLS_PHASESET, Constants.DEFAULT_PHASESET);
+    }
+  }
+
+  public void splitAlternateAlleleInfoFields(String attrName, Object value,
+                                             List<TableRow> altMetadata,
+                                             VCFHeaderLineType type, int count) {
+    if (value instanceof List) {
+      List<Object> valList = (List<Object>)value;
+      if (count != valList.size()) {
+        throw new IndexOutOfBoundsException("Value count does not match the count defined by " +
+            "VCFHeader");
```

Comment:
This method I added the field value one by one into the alt field in a for loop. I think we need to check the size for those two lists. Since adding value one by one will call ```convertSingleValue``` and it does not check the size of the value.